Packs/Syslog/Integrations/Syslogv2/Syslogv2.py

```diff
@@ -0,0 +1,335 @@
+        certificate_file.write(bytes(certificate, 'utf-8'))
+        certificate_file.close()
+
+        private_key_file = NamedTemporaryFile(delete=False)
+        private_key_path = private_key_file.name
+        private_key_file.write(bytes(private_key, 'utf-8'))
+        private_key_file.close()
+        server = StreamServer(('0.0.0.0', port), perform_long_running_execution, keyfile=private_key_path,
+                              certfile=certificate_path)
+        demisto.debug('Starting HTTPS Server')
+    else:
+        server = StreamServer(('0.0.0.0', port), perform_long_running_execution)
+        demisto.debug('Starting HTTP Server')
+    return server
+
+
+def get_mapping_fields() -> Dict[str, str]:
+    def transfer_type_to_str_type(type_: Any):
+        type_as_str: str = str(type_)
+        # remove typing from the type name
+        type_as_str = type_as_str.replace('typing.', '')
+        # For classes other than typing, eg <class 'str'>
+        type_as_str = type_as_str.replace('<class ', '').replace("'", '').replace('>', '')
+        return type_as_str
+
+    return {k: transfer_type_to_str_type(v) for k, v in get_type_hints(SyslogMessageExtract).items()}
+
+
+''' MAIN FUNCTION '''
+
+
+def main() -> None:
+    params = demisto.params()
+    command = demisto.command()
+    message_regex: Optional[str] = params.get('message_regex')
+    incident_type: Optional[str] = params.get('incident_type')
+    certificate: Optional[str] = params.get('certificate')
+    private_key: Optional[str] = params.get('private_key')
+
+    log_format: str = params.get('log_format', '')
+    if log_format not in FORMAT_TO_PARSER_FUNCTION:
+        raise DemistoException(f'Given format: {log_format} is not in the expected format.\n'
+                               f'Please choose one of the following formats: {FORMAT_TO_PARSER_FUNCTION.keys()}')
+
+    try:
+        port = int(params.get('longRunningPort'))
+    except (ValueError, TypeError) as e:
+        raise DemistoException(f'Invalid listen port - {e}') from e
+    if port < 0 or MAX_PORT < port:
+        raise DemistoException(f'Given port: {port} is not valid and must be between 0-{MAX_PORT}')
+
+    demisto.debug(f'Command being called is {demisto.command()}')
+    try:
+        if command == 'test-module':
+            try:
+                prepare_globals_and_create_server(port, log_format, message_regex, incident_type, certificate,
+                                                  private_key)
+            except OSError as e:
+                if 'Address already in use' in str(e):
+                    raise DemistoException(f'Given port: {port} is already in use. Please either change port or '
+                                           f'make sure to close the connection in the server using that port.')
+                raise e
+            return_results('ok')
+        elif command == 'fetch-incidents':
+            fetch_samples()
```

Comment:
when exactly is this happening? let's document and briefly explain the sampling logic